core/state_transition.go

```diff
@@ -555,8 +698,10 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
 		// are 0. This avoids a negative effectiveTip being applied to
 		// the coinbase when simulating calls.
 	} else {
+
 		fee := new(uint256.Int).SetUint64(st.gasUsed())
 		fee.Mul(fee, effectiveTipU256)
+		// always add to balance whether or not gasFromSoul is true
```

Comment:
If the token is not backed by gas token, then we should just burn the token, right?

Reply:
See the above comment.

Reply:
Yes, we'll burn the soul gas token if it's not backed by gas token.